### PR TITLE
Allow for dropout in all decoder layers and fix resetting of rnn layers in decoder.decode().

### PR DESCRIPTION
**Pull request by philschulz**
Before the layers in layer._counter for each rnn layer would grow indefinitely.

### CODE REVIEW

File: sockeye/encoder.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def get_encoder(num_embed: int,
     if num_layers > 1:
         encoders.append(EncoderClass(num_hidden=rnn_num_hidden,
                                      num_layers=num_layers - 1,
-                                     dropout=0.,
```
Comment (fhieber) [inline sockeye/encoder.py]: good catch!

File: sockeye/decoder.py [hunk 2]
```diff
@@ -363,6 +363,8 @@ def decode(self,
         lexical_biases = []
 
         self.rnn.reset()
+        for cell in self.rnn._cells:
```
Comment (fhieber) [inline sockeye/decoder.py]: As noted before, this should ideally be fixed by MXNet so that we don't have to access private members.

Comment (fhieber) [inline sockeye/decoder.py]: Actually, could you add a small TODO comment that we can remove this once MXNet changes?

Comment (philschulz) [inline sockeye/decoder.py]: Done

**Pull request merged**